### PR TITLE
Ajout gestion d'erreur et traitement du caratere EOT

### PR DESCRIPTION
**Pull request by FrColin**
le comptage d'erreur permet de mesurer la fiabilité de la transmission TIC
le caractère EOT est parfois transmis pour interrompre la trame en cours.
Enedis-NOI-CPT_02E.pdf ( Page : 19 )

### CODE REVIEW

File: src/LibTeleinfo.h [hunk 1]
```diff
@@ -142,6 +143,11 @@ class TInfo
     char *        valueGet_P(const char * name, char * value);
     boolean       listDelete();
     unsigned char calcChecksum(char *etiquette, char *valeur, char *horodate=NULL) ;
+    
+    int         getChecksumErrorCount() {return checksumerror;};
+    int         getFrameSizeErrorCount(){return framesizeerror;};
+    int         getFrameFormatErrorCount(){return frameformaterror;};
+    int         getFrameInterruptedCount(){return frameinterrupted;};
```
Comment (hallard) [inline src/LibTeleinfo.h:151]: Could you please set them as uint32_t ?

File: src/LibTeleinfo.h [hunk 2]
```diff
@@ -165,6 +171,11 @@ class TInfo
     void      (*_fn_new_frame)(ValueList * valueslist);
     void      (*_fn_updated_frame)(ValueList * valueslist);
 
+    int       checksumerror;
+    int       frameformaterror;
+    int       framesizeerror;
+    int       frameinterrupted;
+
     //volatile uint8_t *dcport;
```
Comment (hallard) [inline src/LibTeleinfo.h:179]: Could you please set them as uint32_t ?

File: library.properties [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-maintainer=Charles-Henri Hallard <community.hallard.me>
+version=1.1.5
+author=François-Régis Colin
+maintainer=François-Régis Colin
 sentence=Decoder for Teleinfo (aka TIC) from French smart power meters
 paragraph=This is a generic Teleinfo (aka TIC) French Meter Measure Library, it can be used on Arduino, Particle, ESP8266, Raspberry PI or anywhere you can do Cpp coding.
 category=Communication
-url=https://github.com/hallard/LibTeleinfo
+url=https://github.com/FrColin/LibTeleinfo.git
 architectures=*
```
Comment (hallard) [inline library.properties:9]: Would have fair leaving original library author when doing a PR

**Review by hallard — CHANGES_REQUESTED** [review]
Thanks for this PR, really helpfull

File: src/LibTeleinfo.h (hunk 1 above)
Comment (FrColin) [reply src/LibTeleinfo.h:151]: Yes of course you are rigth

File: src/LibTeleinfo.h (hunk 2 above)
Comment (FrColin) [reply src/LibTeleinfo.h:179]: Yes of course you are rigth

File: library.properties (hunk 3 above)
Comment (FrColin) [reply library.properties]: You are extremly rigth a made this change by mistake

**Comment by hallard** [comment]
Thanks for this PR

**Comment by hallard** [comment]
Thank again for this patch looks like now I understand better some errors, check this out
https://community.ch2i.eu/topic/1389/trames-erron%C3%A9es-sur-linky-production-photovolta%C3%AFque/10

If it works I'll update the library into tasmota @barbudor @s-hadinger

**Pull request merged**